code/modules/mob/living/simple_animal/hostile/space_dragon.dm

```diff
@@ -201,10 +206,20 @@
 		destroy_rifts()
 	..()
 	add_dragon_overlay()
+	rift.Remove(src)
+	QDEL_NULL(rift)
+	small_sprite.Remove(src)
+	UnregisterSignal(small_sprite, COMSIG_ACTION_TRIGGER)
+	QDEL_NULL(small_sprite)
```

Comment:
another question: why do you delete all the actions whenever they die, anyways?

Reply:
I can't remember for the life of me tbh